src/gg.regression.unity.bots/Runtime/Scripts/StateRecorder/ScreenRecorder.cs

```diff
@@ -20,6 +21,32 @@
 // ReSharper disable once LoopCanBeConvertedToQuery - Better performance using indexing vs enumerators
 namespace RegressionGames.StateRecorder
 {
+    
+    class TickDataToWriteToDisk
```

Comment:
I see now why this was added, lots of data now. However, I wonder if this is faster as a class or a struct, my C# knowledge is not as strong there, maybe @RG-nAmKcAz would know? It might be faster to create a struct than allocate a new object? https://schwabencode.com/blog/2021/09/20/dotnet-class-vs-struct-performance

Reply:
Yeah we're passing so many variables between multiple methods that adding a new one requires several line changes. I assumed we were passing individual args rather than an object/struct to restrict who can modify values, but if @RG-nAmKcAz's intent is to optimize performance then I'll remove this and accept that we'll just have long method signatures

Reply:
I'll leave this open for Zack to give a final say, but LGTM!